### PR TITLE
update react-instantsearch-dom & core

### PR DESCRIPTION
**Pull request by foray1010**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

1. `appId` and `appKey` no longer recommended way to init the `InstantSearch` (see https://www.algolia.com/doc/api-reference/widgets/instantsearch/react/), it suggests we use `searchClient` now

2. `InfiniteHits` does not have `refine` but `refineNext` and `refinePrevious` (see https://www.algolia.com/doc/api-reference/widgets/infinite-hits/react/?language=react#full-example)

### CODE REVIEW

**Comment by gburgett** [comment]
Thanks @foray1010 !

Regarding `<InstantSearch />`, is it still possible to use `apiKey` and `appId`?  If so I'd prefer we not remove them from the type definitions.  Better just to add the new parameter.

**Comment by foray1010** [comment]
hi @gburgett ! I have read the source code and seems it is no longer possible to use `apiKey` and `appId`
https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-core/src/components/InstantSearch.js#L176-L204

**Comment by gburgett** [comment]
Aha, I figured out the issue!  That component which you've linked is actually not the one exported by `react-instantsearch-dom`.  If you do the following:

```ts
import { InstantSearch } from 'react-instantsearch-dom'
```
then you actually get this object: https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-dom/src/widgets/InstantSearch.js  
which is created by this connector: https://github.com/algolia/react-instantsearch/blob/master/packages/react-instantsearch-core/src/core/createInstantSearch.js  
which wraps that InstantSearch component you have linked above.  The actual prop types therefore ought to be:

```js
static propTypes = {
      algoliaClient: PropTypes.object,
      searchClient: PropTypes.object,
      appId: PropTypes.string,
      apiKey: PropTypes.string,
      children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
      ]),
      indexName: PropTypes.string.isRequired,
      createURL: PropTypes.func,
      searchState: PropTypes.object,
      refresh: PropTypes.bool.isRequired,
      onSearchStateChange: PropTypes.func,
      onSearchParameters: PropTypes.func,
      resultsState: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
      root: PropTypes.shape({
        Root: PropTypes.oneOfType([
          PropTypes.string,
          PropTypes.func,
          PropTypes.object,
        ]).isRequired,
        props: PropTypes.object,
      }),
};
```

**Comment by gburgett** [comment]
@Haroenv can you confirm?

**Review by gburgett — CHANGES_REQUESTED** [review]
I have to explicitly leave a comment with "request changes" - please revert your changes to the InstantSearchProps, or improve them according to the propTypes linked above.

File: types/react-instantsearch-dom/index.d.ts [hunk 1]
```diff
@@ -55,21 +55,19 @@ interface CommonWidgetProps {
 }
 
 export interface InstantSearchProps {
-  apiKey: string;
-  appId: string;
```
Comment (samouss) [inline types/react-instantsearch-dom/index.d.ts]: Those values aren't documented anymore but it's still possible to provide them. Note that they will be removed with the next major version of React InstantSearch. Until the release it's probably better to keep them.

File: types/react-instantsearch-dom/index.d.ts [hunk 2]
```diff
@@ -55,21 +55,19 @@ interface CommonWidgetProps {
...
   indexName: string;
+  searchClient: any;
 
-  algoliaClient?: any;
```
Comment (samouss) [inline types/react-instantsearch-dom/index.d.ts]: This one can be removed though. The `searchClient` is the prefer alternative.

**Review by samouss — CHANGES_REQUESTED** [review]

File: types/react-instantsearch-dom/react-instantsearch-dom-tests.tsx [hunk 3]
```diff
@@ -7,9 +7,8 @@ import { Hit, connectRefinementList, connectMenu } from 'react-instantsearch-cor
 () => {
   const App = () => (
     <InstantSearch
-      appId="latency"
-      apiKey="3d9875e51fbd20c7754e65422f7ce5e1"
```
Comment (samouss) [inline types/react-instantsearch-dom/react-instantsearch-dom-tests.tsx]: We should keep at least one test that use this notation to check that it's actually working.

**Review by samouss — APPROVED** [review]

**Comment by foray1010** [comment]
thank you very much @gburgett and @samouss ! I have marked `searchClient`, `apiKey` and `appId` as optional field

**Review by gburgett — APPROVED** [review]

**Pull request merged**